src/seraphis_impl/serialization_demo_types.h

```diff
@@ -282,12 +283,12 @@ struct ser_LegacyRingSignatureV4 final
 {
     /// a clsag proof
     ser_LegacyClsagProof clsag_proof;
-    /// on-chain indices of the proof's ring members (serializable as index offsets)
-    std::vector<std::uint64_t> reference_set_COMPACT;
+    /// on-chain indices of the proof's ring members
+    std::vector<std::pair<std::uint64_t, std::uint64_t>> reference_set;
```

Comment:
Mainly I got rid of compaction because I'm overhauling serialization in #39 anyways. Here's how I would do compaction:

1. Order pairs by increasing amounts
2. Order each pair for a given amount by increasing index number
3. Decumulate the list of unique amounts
4. Decumulate the list of index numbers for a given amount
5. Store number of unique amounts, then entries of amount offset and a list of index number offets

For example, if your reference set entries `(a_j, i_k)` is `{(0, 0}, (0, 5), (0, 8), (1, 4), (1, 5), (3, 0), (3, 10)}`, the encoding could look like so:
```
3            // number of unique amounts
0            // base amount
3            // number of indexes for amount '0'
0            // base index for amount '0'
5            // index offset
3            // index offset
1            // amount offset
2            // number of indexes for amount '1'
4            // base index for amount '1'
1            // index offset
2            // amount offset
2            // number of indexes for amount '3'
0            // base index for amount '3'
10          // index offset
```

You can see how this can get pretty compact if there are not many unique ledger indexing amounts. For example, if we want to specify the first 20 RingCT enotes for our reference set (ledger indexing amount is 0), then our encoding would be: `1 0 20 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1`.